### PR TITLE
feat: Added doctrine/dbal v4.0.

### PR DESCRIPTION
**Pull request by tasselchof**
fix: #6

### CODE REVIEW

**Comment by leeqvip** [comment]
@tasselchof  Please make it compatible with PHP7.1, it's not very complicated.

**Comment by tasselchof** [comment]
> @tasselchof Please make it compatible with PHP7.1, it's not very complicated.

Except doctrine/dbal:^3.8 requires at least PHP 7.4, so it is impossible to do this. I think PHP 7 should be dropped from supported versions.

File: src/Adapter.php [hunk 1]
```diff
@@ -97,7 +97,7 @@ public static function newAdapter($connection): Adapter
      */
     public function initTable()
     {
-        $sm = $this->connection->getSchemaManager();
+        $sm = $this->connection->createSchemaManager();
```
Comment (leeqvip) [inline src/Adapter.php:100]: You can make some assertions here to ensure compatibility with lower versions.

File: src/Adapter.php [hunk 2]
```diff
@@ -147,7 +147,7 @@ public function savePolicyLine($pType, array $rule)
     public function loadPolicy(Model $model): void
     {
         $queryBuilder = $this->connection->createQueryBuilder();
-        $stmt = $queryBuilder->select('p_type', 'v0', 'v1', 'v2', 'v3', 'v4', 'v5')->from($this->policyTableName)->execute();
+        $stmt = $queryBuilder->select('p_type', 'v0', 'v1', 'v2', 'v3', 'v4', 'v5')->from($this->policyTableName)->executeQuery();
```
Comment (leeqvip) [inline src/Adapter.php:150]: And here

**Review by leeqvip — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
Fix: https://github.com/php-casbin/dbal-adapter/issues/6

**Pull request merged**